### ISSUE TITLE
Syntax Error on valid piece of code using templatefile function

### ISSUE THREAD

**Issue by gokuatkai** [hashicorp/vscode-terraform#214]
Hello,
where
```
resource "aws_iam_policy" "prod_lambda_execution_policy" {
  name = "serverless-${local.project}-${local.environment}-lambda-execution-policy"
  path = "/"
  description = "IAM policy for logging from a lambda"
  policy = "${templatefile(
    "templates/execution_policy.json.tpl",
    {
      account_id = "${local.account_id}"
      region     = "${local.region}"
      project    = "${local.project}"
      kms_key_id = "${local.kms_key_id}"
    }
  )}"
}
```
Is a valid syntax and the extension raise a syntax error: 
```
expected expression but found invalid sequence "{"
Peek Problem
No quick fixes available
```

![image](https://user-images.githubusercontent.com/30238401/65135554-6f17cb00-da06-11e9-9578-85083e056f3d.png)

**Comment by gokuatkai** [comment]
A normal inline syntax also trigger the error in vscode:
```
policy = "${templatefile("templates/creation_policy.json.tpl", {bucket_name = "REDACTED" } )}"
```

**Comment by brandonramsey** [comment]
I'm running into a similar issue -- in my case, with the `merge` function, and using a map variable instead of a string -- but in both cases it looks like the inner `{` may be throwing it off. 

My code:
```hcl
resource "aws_instance" "netscaler" {
  count                  = 1
  ami                    = "${local.netscaler_vendor_ami}"
  instance_type          = "t2.large"
  vpc_security_group_ids = ["${aws_security_group.tenable_automation_sg.id}"]
  tags = "${merge(local.baseTags, {
    Name = "citrix_netscaler"
  })}"
}
```

The error I'm receiving is the same:
<img width="580" alt="image" src="https://user-images.githubusercontent.com/30050495/66707058-bee56b80-ed00-11e9-8660-6b4bfb5c03ad.png">

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
